src/Chart.Core.js

```diff
@@ -1270,8 +1270,13 @@
 				y: chartY
 			});
 
+			//Normalize values to support custom start angle.
+			var startAngle = this.startAngle % (Math.PI * 2),
+				endAngle = this.endAngle % (Math.PI * 2),
+				angle = pointRelativePosition.angle % (Math.PI * 2);
+			endAngle = startAngle > endAngle ? Math.PI * 2 + endAngle : endAngle;
 			//Check if within the range of the open/close angle
```

Comment:
Here I normalized the angles values because the start angle can make a segment go beyond the previous values but we need to wrap the circle values. I tested this pretty hard and it looks pretty solid.